taurus_pyqtgraph/datainspectormodel.py

```diff
@@ -0,0 +1,270 @@
+from datetime import datetime
+
+from pyqtgraph import TextItem, InfiniteLine, PlotDataItem
+from dateaxisitem import DateAxisItem
+
+import numpy as np
+
+
+class DataInspectorModel(object):
+    """
+    DataInspectorModel is a general class use to point-picking.
+    This class use a numpy library to finding the nearest point
+    from the :class:`pyqtgraph.InfiniteLine`.
+    Attach two widget to the plot:
+        - :class:`pyqtgraph.InfiniteLine` - to selecting point
+        - :class:`pyqtgraph.TextItem` - to showing the x,y value
+    Also this class blink the selected scatter point item.
+    """
+    def __init__(self, trigger_point_size=20, default_point_size=10):
+
+        self._scatter = {}
+        self._information_clouds = {}
+        # Maybe good idea is give user's access
+        # to they have possibility to change the cloud style?
+        self.cloud_style = "background-color: #35393C;"
+        self._custom_point_style_init = {}
+
+        self.date_format = "%Y-%m-%d %H:%M:%S"
+
+        self.trigger_point_size = trigger_point_size
+        self.default_point_size = default_point_size
+
+        self._v_line = InfiniteLine(angle=90, movable=False)
+        self.plot = None
+
+    def attachPlot(self, plot):
+        """
+        Method to attach :class:`DataInspectorModel` to the plot
+
+        :param plot: to attach
+        """
+        self.plot = plot
+        self.plot.addItem(self._v_line, ignoreBounds=True)
+
+    def detachPlot(self):
+        """
+        Method use to detach the class:`DataInspectorModel` from the plot
+        """
+        self.detach()
+
+        # Reset the dynamic variables
+        # to allow use the attachToPlot method in the future
+        self._custom_point_style_init = {}
+        self._scatter = {}
+        self._information_clouds = {}
+
+    def mouseMoved(self, evt):
+        """
+        Slot use to handle the mouse move event, and preform the action
+        on the plot.
+
+        :param evt: mouse event
+        """
+        pos = evt[0]
+
+        if self.plot.sceneBoundingRect().contains(pos):
+            mousePoint = self.plot.vb.mapSceneToView(pos)
+
+            # Move InfiniteLine to the new position
+            self._v_line.setPos(mousePoint.x())
+
+            # Iteration over each curves in the plot and checking
+            # that mouse point if is near the x axis date
+            for i, curve in enumerate(filter(lambda o: isinstance(o, PlotDataItem),
+                                             self.plot.curves)):
+
+                curve_data = curve.getData()
+                curve_x_data = curve_data[0]
+                curve_y_data = curve_data[1]
+
+                # Getting the nearest point in the x axis
+                # (if exist - empty array if not)
+                point_index = self._getPointIndex(curve_x_data, mousePoint.x())
```

Comment:
I think that this implementation could be simplified if we used the approach proposed [here](https://groups.google.com/forum/#!msg/pyqtgraph/HU4GsfzAxXU/H8UPYbPjAwAJ) (i.e. basically doing `argmin(xvalues-mousex)`. 

I created [a very basic example](https://gist.github.com/cpascual/5620f913f83cae4be2a4f50057111e4c) which follows that approach and with which we would get rid of the `PointPicker`, `_calculateRoi` and `_getPointIndex`

Reply:
This implementation is great!
I modify the datainspectormodel like in the example and did small changes:

- To prevent shaking the graph on an automatic scale in` _createLabels` I use `setParentItem` method on the `text_item`. In this case pyqtgraph doesn't take into account the text_item and not adjust the scale (this couse the shaking the graph when we pick the point near the border of the plot).

- I add the point trigger logic, and also if needed converting the x value to the time.
Also I fromat the y value to a defined number of decimal places (like in y_format variable).